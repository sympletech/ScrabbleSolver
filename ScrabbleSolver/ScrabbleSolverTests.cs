using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ScrabbleSolver
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ScrabbleSolverTests
    {
        [Test]
        public void TwoTileTest()
        {
            string[] tiles =
                {
                    "a", "b"
                };
            var results = Solver.GetCombinations(tiles.ToList());

            ApprovalTests.Approvals.VerifyAll(results, "foo"); 
        }
        [Test]
        public void TwoTileTest2()
        {
            string[] tiles =
                {
                    "c", "d"
                };
            var results = Solver.GetCombinations(tiles.ToList());

            ApprovalTests.Approvals.VerifyAll(results, "foo");
        }
        [Test]
        public void TwoTileTest3()
        {
            string[] tiles =
                {
                    "a","b","c", "d"
                };
            var results = Solver.GetCombinations(tiles.ToList());

            ApprovalTests.Approvals.VerifyAll(results, "foo");
        }

        [Test]
        public void ScoreAbcTest()
        {
            var word = new List<string>(){"abc", "def"};
            var results = new Dictionary<string, int>();
            
            foreach (var w in word)
            {
                results[w]=Solver.GetScore(w);
            }
            ApprovalTests.Approvals.VerifyAll(results);
        }

        [Test]
        public void GetScoresForTilesTest()
        {
            string[] tiles =
                {
                    "a","b","c", "d"
                };
            var results = Solver.GetScoredAndRankedCombinations(tiles.ToList());

             ApprovalTests.Approvals.VerifyAll(results, "doo");

        }

    }
}
