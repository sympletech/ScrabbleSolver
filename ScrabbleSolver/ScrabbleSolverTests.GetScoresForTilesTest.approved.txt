doo[0] = [abcd, 9]
doo[1] = [abdc, 9]
doo[2] = [acbd, 9]
doo[3] = [acdb, 9]
doo[4] = [adbc, 9]
doo[5] = [adcb, 9]
doo[6] = [bacd, 9]
doo[7] = [badc, 9]
doo[8] = [bcad, 9]
doo[9] = [bcda, 9]
doo[10] = [bdac, 9]
doo[11] = [bdca, 9]
doo[12] = [cabd, 9]
doo[13] = [cadb, 9]
doo[14] = [cbad, 9]
doo[15] = [cbda, 9]
doo[16] = [cdab, 9]
doo[17] = [cdba, 9]
doo[18] = [dabc, 9]
doo[19] = [dacb, 9]
doo[20] = [dbac, 9]
doo[21] = [dbca, 9]
doo[22] = [dcab, 9]
doo[23] = [dcba, 9]
doo[24] = [bcd, 8]
doo[25] = [bdc, 8]
doo[26] = [cbd, 8]
doo[27] = [cdb, 8]
doo[28] = [dbc, 8]
doo[29] = [dcb, 8]
doo[30] = [abc, 7]
doo[31] = [acb, 7]
doo[32] = [bac, 7]
doo[33] = [bca, 7]
doo[34] = [cab, 7]
doo[35] = [cba, 7]
doo[36] = [abd, 6]
doo[37] = [acd, 6]
doo[38] = [adb, 6]
doo[39] = [adc, 6]
doo[40] = [bad, 6]
doo[41] = [bc, 6]
doo[42] = [bda, 6]
doo[43] = [cad, 6]
doo[44] = [cb, 6]
doo[45] = [cda, 6]
doo[46] = [dab, 6]
doo[47] = [dac, 6]
doo[48] = [dba, 6]
doo[49] = [dca, 6]
doo[50] = [bd, 5]
doo[51] = [cd, 5]
doo[52] = [db, 5]
doo[53] = [dc, 5]
doo[54] = [ab, 4]
doo[55] = [ac, 4]
doo[56] = [ba, 4]
doo[57] = [ca, 4]
doo[58] = [ad, 3]
doo[59] = [b, 3]
doo[60] = [c, 3]
doo[61] = [da, 3]
doo[62] = [d, 2]
doo[63] = [a, 1]
